www/src/pages/table/NodeSearchComponent.tsx

```diff
@@ -44,62 +55,115 @@ export const NodeSearchComponent = withStyles(styles)(
       classes,
+      classAndLiteralSearchOnly,
     }: {
       sm: SemanticModel;
       value?: SearchValue;
       onDeselect: (value: SearchValue) => void;
       onSelect: (value: SearchValue) => void;
+      classAndLiteralSearchOnly: boolean;
     } & WithStyles<typeof styles>) => {
       const { classStore } = useStores();
-      const seqInvoker = new SequentialFuncInvoker();
+      const [searchOptions, setSearchOptions] = useState<SearchOptions[]>();
 
       // gather all options already in the store, leverage the fact
       // that property store is readonly
       const options = useMemo(() => {
-        const options: ({
-          value: string;
-          label: string;
-        } & SearchValue)[] = [];
-        for (const r of classStore.iter()) {
-          options.push({
-            type: "class",
-            id: r.id,
-            value: `class:${r.id}`,
-            label: sm.graph.uriCount.nextLabel(r.uri, r.readableLabel),
-          });
-        }
+        const options: SearchOptions[] = [];
 
         for (const u of sm.graph.nodes) {
+          if (
+            classAndLiteralSearchOnly &&
+            !(u.nodetype == "class_node" || u.nodetype == "literal_node")
+          )
+            continue;
           options.push({
             type: u.nodetype,
             id: u.id,
             value: `${u.nodetype}:${u.id}`,
             label: sm.graph.uriCount.label(u),
+            filterlabel: sm.graph.uriCount.label(u),
             className: classes[u.nodetype],
-          } as any);
+          });
         }
+
+        setSearchOptions(options);
         return options;
-      }, [classStore.records.size]);
+      }, [sm.graph.version]);
 
       // search for additional values if it's not in the list
       const onSearch = (query: string) => {
-        if (query === "") return;
-        seqInvoker.exec(() => {
-          return classStore.fetchById(query).then(() => 1);
+        if (query === "") {
+          setSearchOptions(options);
+          return;
+        }
+        const loaderOption: SearchOptions = {
+          type: "class",
+          id: "",
+          label: <Spin style={{ width: "100%", marginTop: 3 }} size="large" />,
+          filterlabel: query,
+          value: "",
+          className: "",
+        };
+
+        console.log(options);
+        setSearchOptions([...options, loaderOption]);
+
+        classStore.findByName(query).then((data) => {
+          let searchResults: SearchOptions[] = data.map(
+            (searchResult: ClassTextSearchResult) => {
+              return {
+                type: "class",
+                id: searchResult.id,
+                label: (
+                  <SearchOptionsComponent
+                    id={searchResult.id}
+                    description={searchResult.description}
+                    label={searchResult.label}
+                  />
+                ),
+                filterlabel: `${searchResult.label} (${searchResult.id})`,
```

Comment:
still have filterLabel

Reply:
we need to have filterlabel to sort
our label is a react component